spec/services/complete_moab_handler_create_spec.rb

```diff
@@ -52,6 +52,23 @@
       expect(results).to include(a_hash_including(code => a_string_matching('CompleteMoab db object already exists')))
     end
 
+    context 'a moab (for the same druid) on a different storage root already exists' do
+      before do
+        po = create(:preserved_object, druid: druid)
+        other_storage_root = create(:moab_storage_root)
+        create(:complete_moab, preserved_object: po, moab_storage_root: other_storage_root)
+      end
+
+      it 'creates the new CompleteMoab in database' do
+        complete_moab_handler.create
+        existing_po = PreservedObject.find_by(druid: druid)
+        new_cm = existing_po.complete_moabs.find_by(moab_storage_root: ms_root)
+        expect(existing_po.current_version).to eq incoming_version
```

Comment:
i believe this test would pass if this line were changed to be

```ruby
expect(existing_po.current_version).to eq existing_cm.version
```

and if the `before` block were turned into 3 `let` statements (or whatever similar approach):
```ruby
let(:po) { create(:preserved_object, druid: druid) }
let(:other_storage_root) { create(:moab_storage_root) }
let(:existing_cm) { create(:complete_moab, preserved_object: po, moab_storage_root: other_storage_root) }
```